aten/src/ATen/cuda/CUDAGeneratorImpl.h

```diff
@@ -87,9 +92,34 @@ namespace at {
  *
  */
 
+struct CUDAGeneratorState : public c10::intrusive_ptr_target {
+  uint64_t seed_;
+  uint64_t philox_offset_per_thread_;
+  uint32_t offset_intragraph_;
+  bool capturing_{};
+  bool is_gpu_tensor_allocated_{};
+  std::unique_ptr<at::Tensor> seed_extragraph_{};
+  std::unique_ptr<at::Tensor> offset_extragraph_{};
```

Comment:
Looks like the compiler doesn't like using `std::unique_ptr` here:
```
/usr/include/c++/9/bits/unique_ptr.h(79): error: incomplete type is not allowed
   static_assert(sizeof(_Tp)>0,
                        ^
          detected during:
            instantiation of "void std::default_delete<_Tp>::operator()(_Tp *) const [with _Tp=at::Tensor]" at line 292
            instantiation of "std::unique_ptr<_Tp, _Dp>::~unique_ptr() noexcept [with _Tp=at::Tensor, _Dp=std::default_delete<at::Tensor>]" at line 101 of /var/lib/jenkins/workspace/aten/src/ATen/cuda/CUDAGeneratorImpl.h
```
Since I assume the `at::Tensor` here is really just an unused wrapper for the storage, is there any issue over just paying the overhead of storing the reference `at::Tensor` rather than a `unique_ptr` to it?

Reply:
Just saw the above comment on `TORCH_ASSERT_NO_OPERATORS`, let me check if there are other places where `at::Tensor` is used in a similar way in a header

Reply:
For now we can try working around with `#undef TORCH_ASSERT_NO_OPERATORS` before including `ATen/core/Tensor.h` to run CI

Reply:
Somehow, this did not trigger a compile error in my local build 🤣. Ideally, `seed_extragraph_` should be of type `at::Tensor`. Direct usage of `at::Tensor` in the header file triggers the `TORCH_ASSERT_NO_OPERATORS` error, which is why I initially used `std::unique_ptr`. However, I have discovered that using `TensorBase` instead can resolve this issue. I have updated the pull request accordingly.